rclpy/src/rclpy/node.cpp

```diff
@@ -16,19 +16,54 @@
 #include <pybind11/pybind11.h>
 
 #include <rcl/error_handling.h>
+#include <rcl/graph.h>
 #include <rcl/rcl.h>
 
 #include <string>
 
-#include "rclpy_common/handle.h"
-
 #include "rclpy_common/exceptions.hpp"
+#include "rclpy_common/handle.h"
 
 #include "node.hpp"
+#include "utils.hpp"
 
 
 namespace rclpy
 {
+const char *
+get_node_fully_qualified_name(py::capsule pynode)
+{
+  auto node = static_cast<rcl_node_t *>(
+    rclpy_handle_get_pointer_from_capsule(pynode.ptr(), "rcl_node_t"));
+  if (!node) {
+    throw py::error_already_set();
+  }
+
+  const char * fully_qualified_node_name = rcl_node_get_fully_qualified_name(node);
+  if (!fully_qualified_node_name) {
+    throw RCLError("Fully qualified name not set");
```

Comment:
@gbalke why change the API? I see the previous implementation used to return `None` (which you can do by changing the return value to `py::object` and returning `py::none()` in this case). Same elsewhere.

Reply:
I was doing this given @sloretz commentary https://github.com/ros2/rclpy/pull/719#discussion_r596448711

Reply:
Oh, I see. Nevermind my comment then.